src/backend/common/helpers/matchstats_helper.py

```diff
@@ -8,29 +8,146 @@
 
 # x is OPR and should be n x 1
 
-from collections import defaultdict
-from typing import Any, Dict, List, Optional, Tuple
+from collections import OrderedDict
+from typing import Callable, Dict, List, Tuple
 
 import numpy as np
-from google.cloud import ndb
 from pyre_extensions import none_throws
 
-from backend.common.consts.alliance_color import ALLIANCE_COLORS, AllianceColor
+from backend.common.consts.alliance_color import (
+    ALLIANCE_COLORS,
+    AllianceColor,
+    OPPONENT,
+)
 from backend.common.consts.comp_level import CompLevel
-from backend.common.consts.event_type import EventType
-from backend.common.memcache import MemcacheClient
-from backend.common.models.keys import TeamId, Year
+from backend.common.models.event_matchstats import (
+    Component,
+    EventMatchstats,
+    StatType,
+    TeamStatMap,
+)
+from backend.common.models.keys import TeamId
 from backend.common.models.match import Match
-from backend.common.models.stats import EventMatchStats, StatType
-from backend.common.queries import event_query
-
 
+StatAccessor = Callable[[Match, AllianceColor], float]
 TTeamIdMap = Dict[TeamId, int]
 
 
-class MatchstatsHelper(object):
+class MatchstatsHelper:
+    OPR_ACCESSOR: StatAccessor = lambda match, color: match.alliances[color]["score"]
+    DPR_ACCESSOR: StatAccessor = lambda match, color: match.alliances[OPPONENT[color]][
+        "score"
+    ]
+    CCWM_ACCESSOR: StatAccessor = lambda match, color: (
+        match.alliances[color]["score"] - match.alliances[OPPONENT[color]]["score"]
+    )
+    DEFAULT_COMPONENT_ACCESSOR: Callable[[Component], StatAccessor] = lambda comp: (
+        lambda match, color: match.score_breakdown[color].get(comp, 0)
+    )
+
+    EVERGREEN: Dict[StatType, StatAccessor] = {
+        StatType.OPR: OPR_ACCESSOR,
+        StatType.DPR: DPR_ACCESSOR,
+        StatType.CCWM: CCWM_ACCESSOR,
+    }
+
+    COMPONENTS = {
+        2020: {
+            "Total Power Cell Points": lambda match, color: (
+                match.score_breakdown[color].get("autoCellPoints")
+                + match.score_breakdown[color].get("teleopCellPoints")
+            ),
+        },
+        2019: {
+            "Cargo + Panel Points": lambda match, color: (
+                match.score_breakdown[color].get("cargoPoints")
+                + match.score_breakdown[color].get("hatchPanelPoints")
+            )
+        },
+    }
+
+    @classmethod
+    def calculate_matchstats(
+        cls, matches: List[Match], skip_coprs: bool = False, keyed: bool = True
```

Comment:
The only place I'm seeing us pass this `keyed` variable is in tests. I feel like we should just always key these? Removing the complexity of this method feels helpful.